test/e2e/index.ts

```diff
@@ -1441,6 +1461,82 @@ test(
   })
 );
 
+test(
+  "client reconnects websocket if closed before making a zome call",
+  withConductor(ADMIN_PORT, async (t) => {
+    const { cell_id, client, admin } = await installAppAndDna(
+      ADMIN_PORT,
+      false,
+      0
+    );
+    await admin.authorizeSigningCredentials(cell_id);
+    await client.client.close();
+    const callParams = {
+      cell_id,
+      zome_name: TEST_ZOME_NAME,
+      fn_name: "bar",
+      provenance: cell_id[1],
+      payload: null,
+    };
+    try {
+      await client.callZome(callParams);
+      t.pass("websocket was reconnected successfully");
+    } catch (error) {
+      t.fail(`websocket was not reconnected: ${error}`);
+    }
+  })
+);
+
+test(
+  "client fails to reconnect to websocket if closed before making a zome call if the provided token is invalid",
+  withConductor(ADMIN_PORT, async (t) => {
+    const { cell_id, client, admin } = await installAppAndDna(ADMIN_PORT);
+    await admin.authorizeSigningCredentials(cell_id);
+    await client.client.close();
+    const callParams = {
+      cell_id,
+      zome_name: TEST_ZOME_NAME,
+      fn_name: "bar",
+      provenance: cell_id[1],
+      payload: null,
+    };
+
+    // Websocket is closed and app authentication token has expired. Websocket reconnection
+    // should fail.
+    try {
+      await client.callZome(callParams, 3000);
+      t.fail(
+        "reconnecting to websocket should have failed due to an invalid token."
+      );
+    } catch (error) {
+      t.assert(
+        error instanceof HolochainError,
+        "error should be of type HolochainError"
+      );
+      assert(error instanceof HolochainError);
+      t.equal(error.name, "InvalidTokenError", "expected an InvalidTokenError");
```

Comment:
For the future - create an enum with all possible error kinds https://github.com/holochain/holochain-client-js/issues/281